pennylane_lightning/src/tests/Test_TypeTraits.cpp

```diff
@@ -0,0 +1,53 @@
+#include "TypeTraits.hpp"
+
+#include <catch2/catch.hpp>
+
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+using namespace Pennylane::Util;
+
+TEST_CASE("Test remove_complex") {
+    SECTION("remove_complex returns the floating point if the given type is "
+            "std::complex") {
+        STATIC_REQUIRE(
+            std::is_same_v<remove_complex_t<std::complex<float>>, float>);
+        STATIC_REQUIRE(
+            std::is_same_v<remove_complex_t<std::complex<double>>, double>);
+    }
+    SECTION("remove_complex returns the same type if not") {
+        STATIC_REQUIRE(std::is_same_v<remove_complex_t<float>, float>);
+        STATIC_REQUIRE(std::is_same_v<remove_complex_t<double>, double>);
+    }
+}
+
+TEST_CASE("Test is_complex") {
+    SECTION("is_complex returns true if the given type is std::complex") {
+        STATIC_REQUIRE(is_complex_v<std::complex<double>>);
+        STATIC_REQUIRE(is_complex_v<std::complex<float>>);
+    }
+    SECTION("remove_complex returns false if not") {
+        STATIC_REQUIRE(!is_complex_v<int>);
+        STATIC_REQUIRE(!is_complex_v<long>);
+        STATIC_REQUIRE(!is_complex_v<float>);
+        STATIC_REQUIRE(!is_complex_v<double>);
+    }
+}
+
+std::pair<int, int> g(std::tuple<int, int, int>);
+
+TEST_CASE("Test FuncReturn") {
+    SECTION("FuncReturn returns correctly returns the return type") {
```

Comment:
Sorry, this sentence is giving me a headache to decypher.

Reply:
Oops sorry :cry: Will fix it

Reply:
Fixed.

Reply:
Did you push your changes here?

Reply:
https://github.com/PennyLaneAI/pennylane-lightning/pull/404/files#diff-7f06705bf867093ae4bef43e10928a7d7a66329a700f1c9167d5cc3ce7700aefR41 Yeap it seems like GitHub does not correctly show changes.

Reply:
Thank you! It is perfect now!